File: Java/1160. Find Words That Can Be Formed by Characters.java
class Solution {
    public int countCharacters(String[] words, String chars) {
        int[] count = new int[26];

        for (char c: chars.toCharArray()) {
            count[c - 'a']++;
        }

        int ans = 0;
        for (String w: words) {
            int[] word = new int[26];
            boolean flag = true;
            for (char c: w.toCharArray()) {
                word[c - 'a']++;
                if (word[c - 'a'] > count[c - 'a']) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                ans += w.length();
            }
        }

        return ans;
    }
}

File: Python3/1160. Find Words That Can Be Formed by Characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        count = Counter(chars)
        ans = 0

        for w in words:
            word = Counter()
            flag = True
            for c in w:
                word[c] += 1
                if word[c] > count[c]:
                    flag = False
                    break
                
            if flag:
                ans += len(w)

        return ans